repobee_sanitizer/_sanitize_file.py

```diff
@@ -24,16 +24,27 @@ def _sanitize_file(
         """A callback function that runs the sanitization protocol on a given
         file.
 
-        Args:
-            args: Parsed and processed args from the RepoBee CLI.
-            api: A platform API instance.
-        Returns:
-            A mapping (str -> List[plug.Result]) that RepoBee's CLI can use for
-            output.
+        Args: args: Parsed and processed args from the RepoBee CLI. api: A
+            platform API instance.
+
+        Returns: An optional plug.PlugResult if the syntax is invalid,
+            otherwise nothing.
         """
+        errors = _syntax.check_syntax(args.infile.read_text().split("\n"))
+        if errors:
+            msg = []
+            for error in errors:
+                msg.append(f"Line {error.line}: {error.msg}\n")
+
```

Comment:
You have an error formatting function in `_sanitize_repo`. Make that a public function in a new module and reuse it instead of doing the same thing again here.

Reply:
It has its own because I thought reporting how many files have an error is silly when you sanitize 1 file. Also telling the user what file is also silly since they had to specify that file themselves.

Reply:
So the change feels unnecessary for the work of creating a new module. Also if we're going to change how we report errors in the future (which we have talked about), we can do that then

Reply:
So what happens e.g. with global errors in this single-file error formatting? Is that handled?

Reply:
Moving one function that is used in one place to another module is something that takes roughly 10 seconds, so I don't see how it's a lot of work. I would strongly advice you to do this refactor, because not only does this duplicate the error output formatting from sanitize repo, it does so in a slightly more buggy way.

Reply:
Reasonable, ill change it

Reply:
Cool. As you've probably noticed by now, nice output formatting is far from trivial. Corner cases abound, and it's really quite difficult to get it "just right". Therefore, it's something you really don't want to do twice :)